File: algorithms/algorithms-cplusplus/leetcode/Question_0492_Construct_the_Rectangle.cc
#include <math.h>
#include <vector>

using namespace std;

class Solution {
 public:
  vector<int> constructRectangle(int area) {
    int r = sqrt(area);
    while (area % r != 0) {
      --r;
    }
    return {area / r, r};
  }
};

File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0492_Construct_the_Rectangle.java
package com.lanceolata.coding.algorithms.java.leetcode;

public class Question_0492_Construct_the_Rectangle {
    public int[] constructRectangle(int area) {
        int i = (int)Math.sqrt(area);
        while (area % i != 0) {
            i--;
        }
        return new int[] {area / i, i};
    }
}
